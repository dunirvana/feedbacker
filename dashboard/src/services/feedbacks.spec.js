import mockAxios from 'axios'
import FeedbacksService from './feedbacks'

jest.mock('axios')

describe('FeedbacksService', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should return all feedbacks', async () => {
    const feedback1 = {
        text: 'Feedback 1',
        id: '1234',
        type: 'IDEA',
        createAt: '1608681600000'
      }
    const feedback2 = {
        text: 'Feedback 2',
        id: '56789',
        type: 'OTHER',
        createAt: '1605225600000'
      }      
    const feedbacks = [ feedback1, feedback2 ]

    mockAxios.get.mockImplementationOnce(() => {
      return Promise.resolve({ data: feedbacks })
    })

    const response = await FeedbacksService(mockAxios).getAll()
    expect(response.data).toContain(feedback1)
    expect(response.data).toContain(feedback2)
    expect(response).toMatchSnapshot()
  })
  
  it('should return feedbacks summary', async () => {
    const summary = {
        all: '2',
        issue: '3',
        idea: '4',
        other: '5'
      }

    mockAxios.get.mockImplementationOnce(() => {
      return Promise.resolve({ data: summary })
    })

    const response = await FeedbacksService(mockAxios).getSummary()
    expect(response.data).toHaveProperty('all')
    expect(response.data).toHaveProperty('issue')
    expect(response.data).toHaveProperty('idea')
    expect(response.data).toHaveProperty('other')
    expect(response).toMatchSnapshot()
  })  
})

