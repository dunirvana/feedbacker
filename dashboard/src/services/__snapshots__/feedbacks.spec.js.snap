// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbacksService should return all feedbacks 1`] = `
Object {
  "data": Array [
    Object {
      "createAt": "1608681600000",
      "id": "1234",
      "text": "Feedback 1",
      "type": "IDEA",
    },
    Object {
      "createAt": "1605225600000",
      "id": "56789",
      "text": "Feedback 2",
      "type": "OTHER",
    },
  ],
}
`;

exports[`FeedbacksService should return feedbacks summary 1`] = `
Object {
  "data": Object {
    "all": "2",
    "idea": "4",
    "issue": "3",
    "other": "5",
  },
}
`;
